csvcubed/tests/unit/cli/inspect/test_inspectsparqlqueries.py

```diff
@@ -0,0 +1,150 @@
+    )
+    assert info_dict.get("licence") is None
+    assert (
+        str(info_dict.get("creator"))
+        == "https://www.gov.uk/government/organisations/hm-revenue-customs"
+    )
+    assert (
+        str(info_dict.get("publisher"))
+        == "https://www.gov.uk/government/organisations/hm-revenue-customs"
+    )
+    assert (
+        len(info_dict["landingPages"].split("|")) == 1
+        and info_dict["landingPages"].split("|")[0]
+        == "https://www.gov.uk/government/statistics/alcohol-bulletin"
+    )
+    assert (
+        len(info_dict["themes"].split("|")) == 1
+        and info_dict["themes"].split("|")[0] == "http://gss-data.org.uk/def/gdp#trade"
+    )
+    assert len(info_dict["keywords"]) == 0
+    assert info_dict.get("contactPoint") is None
+    assert str(info_dict["identifier"]) == "Alcohol Bulletin"
+
+
+def test_select_csvw_information_for_codelist():
+    csvw_metadata_json_path = _test_case_base_dir / "codelist.csv-metadata.json"
+    metadata_processor = MetadataProcessor(csvw_metadata_json_path)
+    csvw_metadata_rdf_graph = metadata_processor.load_json_ld_to_rdflib_graph()
+
+    result = select_csvw_information(csvw_metadata_rdf_graph)
+    info_dict = result.asdict()
+
+    assert str(info_dict["title"]) == "Alcohol Content"
+    assert str(info_dict["label"]) == "Alcohol Content"
+    assert (
+        dateutil.parser.isoparse(info_dict["issued"]).strftime("%Y-%m-%d %H:%M:%S.%f%Z")
+        == "2022-02-11 21:00:21.040987"
+    )
+    assert (
+        dateutil.parser.isoparse(info_dict["modified"]).strftime(
+            "%Y-%m-%d %H:%M:%S.%f%Z"
+        )
+        == "2022-02-11 21:00:21.040987"
+    )
+    assert info_dict.get("comment") is None
+    assert info_dict.get("description") is None
+    assert info_dict.get("license") is None
+    assert info_dict.get("creator") is None
+    assert info_dict.get("publisher") is None
+    assert len(info_dict["landingPages"]) == 0
+    assert len(info_dict["themes"]) == 0
+    assert len(info_dict["keywords"]) == 0
+    assert info_dict.get("contactPoint") is None
+    assert str(info_dict.get("identifier")) == "Alcohol Content"
+
+
+def test_select_csvw_dsd_dataset():
+    csvw_metadata_json_path = _test_case_base_dir / "datacube.csv-metadata.json"
+    metadata_processor = MetadataProcessor(csvw_metadata_json_path)
+    csvw_metadata_rdf_graph = metadata_processor.load_json_ld_to_rdflib_graph()
+
+    result = select_csvw_dsd_dataset_label_and_dsd_def_uri(csvw_metadata_rdf_graph)
+    result_dict = result.asdict()
+    components = select_csvw_dsd_qube_components(
+        csvw_metadata_rdf_graph, result_dict["dataStructureDefinition"]
+    )
+
+    assert (
+        str(result_dict["dataStructureDefinition"])
+        == "file:///workspaces/csvcubed/alcohol-bulletin.csv#structure"
+    )
+    assert str(result_dict["dataSetLabel"]) == "Alcohol Bulletin"
+    assert len(components) == 17
```

Comment:
It would probably be useful to do a bit more in-depth testing that all of the appropriate fields come back here and that optional fields work as expected too. 

I wonder whether it might be easier to write a class to hold the information with clearly defined types instead of relying on dictionaries?